src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/NullableConverterFactory.cs

```diff
@@ -46,6 +51,8 @@ public static JsonConverter CreateValueConverter(Type valueTypeToConvert, JsonCo
         [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2055:MakeGenericType",
             Justification = "'NullableConverter<T> where T : struct' implies 'T : new()', so the trimmer is warning calling MakeGenericType here because valueTypeToConvert's constructors are not annotated. " +
             "But NullableConverter doesn't call new T(), so this is safe.")]
+        [UnconditionalSuppressMessage("AotAnalysis", "IL3050:RequiresDynamicCode", Justification = "The factory constructors are only invoked in the context of reflection serialization code paths " +
```

Comment:
Same comment here as https://github.com/dotnet/runtime/pull/68464#discussion_r859954547. This should be marked with `RequiresDynamicCode` to ensure all callers are appropriately annotated.

Reply:
Thanks! Moved to class level